From 8d95631ea6c039a60bb7ac456f687a8fdf0c4381 Mon Sep 17 00:00:00 2001
From: Filipe Brandenburger <filbranden@google.com>
Date: Mon, 25 Aug 2014 22:05:03 -0700
Subject: [PATCH] test-path-util: use assert_se in all assertions

Otherwise they get optimized out when CPPFLAGS='-DNDEBUG' is used, and that
causes the tests to fail.

Tested:
- make check TESTS='test-path-util' CPPFLAGS='-DNDEBUG'
---
 src/test/test-path-util.c | 30 +++++++++++++++---------------
 1 file changed, 15 insertions(+), 15 deletions(-)

diff --git a/src/test/test-path-util.c b/src/test/test-path-util.c
index c8dcd85397..01afb3e6fe 100644
--- a/src/test/test-path-util.c
+++ b/src/test/test-path-util.c
@@ -79,35 +79,35 @@ static void test_path(void) {
                 char p2[] = "//aaa/.////ccc";
                 char p3[] = "/./";
 
-                assert(path_equal(path_kill_slashes(p1), "aaa/bbb/ccc"));
-                assert(path_equal(path_kill_slashes(p2), "/aaa/./ccc"));
-                assert(path_equal(path_kill_slashes(p3), "/./"));
+                assert_se(path_equal(path_kill_slashes(p1), "aaa/bbb/ccc"));
+                assert_se(path_equal(path_kill_slashes(p2), "/aaa/./ccc"));
+                assert_se(path_equal(path_kill_slashes(p3), "/./"));
         }
 }
 
 static void test_find_binary(const char *self) {
         char *p;
 
-        assert(find_binary("/bin/sh", &p) == 0);
+        assert_se(find_binary("/bin/sh", &p) == 0);
         puts(p);
-        assert(streq(p, "/bin/sh"));
+        assert_se(streq(p, "/bin/sh"));
         free(p);
 
-        assert(find_binary(self, &p) == 0);
+        assert_se(find_binary(self, &p) == 0);
         puts(p);
-        assert(endswith(p, "/test-path-util"));
-        assert(path_is_absolute(p));
+        assert_se(endswith(p, "/test-path-util"));
+        assert_se(path_is_absolute(p));
         free(p);
 
-        assert(find_binary("sh", &p) == 0);
+        assert_se(find_binary("sh", &p) == 0);
         puts(p);
-        assert(endswith(p, "/sh"));
-        assert(path_is_absolute(p));
+        assert_se(endswith(p, "/sh"));
+        assert_se(path_is_absolute(p));
         free(p);
 
-        assert(find_binary("xxxx-xxxx", &p) == -ENOENT);
+        assert_se(find_binary("xxxx-xxxx", &p) == -ENOENT);
 
-        assert(find_binary("/some/dir/xxxx-xxxx", &p) == -ENOENT);
+        assert_se(find_binary("/some/dir/xxxx-xxxx", &p) == -ENOENT);
 }
 
 static void test_prefixes(void) {
@@ -156,8 +156,8 @@ static void test_prefixes(void) {
 
         b = false;
         PATH_FOREACH_PREFIX_MORE(s, "") {
-                assert(!b);
-                assert(streq(s, ""));
+                assert_se(!b);
+                assert_se(streq(s, ""));
                 b = true;
         }
 }
